audio/src/constant_source.rs

```diff
@@ -0,0 +1,70 @@
+use block::Chunk;
+use block::Tick;
+use node::AudioNodeEngine;
+use node::BlockInfo;
+use node::{AudioNodeType, ChannelInfo};
+use param::{Param, ParamType};
+
+#[derive(Copy, Clone, Debug)]
+pub struct ConstantSourceNodeOptions {
+    pub offset: f32,
+}
+
+impl Default for ConstantSourceNodeOptions {
+    fn default() -> Self {
+        ConstantSourceNodeOptions { offset: 1. }
+    }
+}
+
+#[derive(AudioNodeCommon)]
+pub(crate) struct ConstantSourceNode {
+    channel_info: ChannelInfo,
+    offset: Param,
+}
+
+impl ConstantSourceNode {
+    pub fn new(options: ConstantSourceNodeOptions, channel_info: ChannelInfo) -> Self {
+        Self {
+            channel_info,
+            offset: Param::new(options.offset),
+        }
+    }
+
+    pub fn update_parameters(&mut self, info: &BlockInfo, tick: Tick) -> bool {
+        self.offset.update(info, tick)
+    }
+}
+
+impl AudioNodeEngine for ConstantSourceNode {
+    fn node_type(&self) -> AudioNodeType {
+        AudioNodeType::ConstantSourceNode
+    }
+
+    fn process(&mut self, mut inputs: Chunk, info: &BlockInfo) -> Chunk {
+        debug_assert!(inputs.len() == 1);
+        inputs.blocks[0].explicit_silence();
+        if inputs.blocks[0].is_silence() {
```

Comment:
This will always be true given that you call `explicit_silence` in the previous line. But in any case, I don't think you need this check and/or early return here. We will have silence at this point for sure, what we need to do here is to fill that silence with audio data :)